lucene/core/src/java/org/apache/lucene/search/TopDocsCollector.java

```diff
@@ -136,14 +136,21 @@ public TopDocs topDocs(int start, int howMany) {
     // pq.size() or totalHits.
     int size = topDocsSize();
 
-    // Don't bother to throw an exception, just return an empty TopDocs in case
-    // the parameters are invalid or out of range.
-    // TODO: shouldn't we throw IAE if apps give bad params here so they dont
-    // have sneaky silent bugs?
-    if (start < 0 || start >= size || howMany <= 0) {
+
+    if (start < 0) {
+      throw new IllegalArgumentException("Expected value of starting position is between 0 and " + size +
```

Comment:
The message should say between "0 and $howMany" rather than size I think, since howMany is the user-supplied parameter and size depends on the number of matches, which the user has no control on?

Reply:
```suggestion
      throw new IllegalArgumentException("Expected value of starting position is between 0 and " + howMany +
```